metricflow/dataflow/nodes/transform_time_dimensions.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+from abc import ABC
+from dataclasses import dataclass
+from typing import Sequence
+
+from metricflow_semantics.dag.id_prefix import IdPrefix, StaticIdPrefix
+from metricflow_semantics.dag.mf_dag import DisplayedProperty
+from metricflow_semantics.specs.time_dimension_spec import TimeDimensionSpec
+from metricflow_semantics.visitor import VisitorOutputT
+
+from metricflow.dataflow.dataflow_plan import DataflowPlanNode
+from metricflow.dataflow.dataflow_plan_visitor import DataflowPlanNodeVisitor
+
+
+@dataclass(frozen=True, eq=False)
+class TransformTimeDimensionsNode(DataflowPlanNode, ABC):
```

Comment:
Maybe: make this an aliasing node (more generic) - a dict of generic specs to specs; input specs / output specs, something like that